File: src/typescript/structure/organisation/statement-issues/processOrder.ts
interface Item {
    name: string;
    price: number;
    quantity: number;
}

interface Order {
    id: string;
    items: Item[];
    discountCode?: string;
}

function processOrder(order: Order, taxRate: number): number {
    let total = (order.items.reduce((sum, item) => sum + item.price * item.quantity, 0) -
        (order.discountCode ? 50 : 0)) * (1 + taxRate);
    console.log(`Order ${order.id} processed. Total after tax: $${total.toFixed(2)}`);
    return total;
}

const order: Order = {
    id: "ORD123",
    items: [
        { name: "Laptop", price: 999.99, quantity: 1 },
        { name: "Mouse", price: 49.99, quantity: 2 },
        { name: "Keyboard", price: 79.99, quantity: 1 },
    ],
    discountCode: "SUMMER21",
};
const taxRate = 0.08;
const total = processOrder(order, taxRate);
console.log(total);
\ No newline at end of file

File: truth_set/go/structure/organisation/statement-issues/processOrder.go
package main

import "fmt"

type Item struct {
	Name     string
	Price    float64
	Quantity int
}

type Order struct {
	ID           string
	Items        []Item
	DiscountCode string
}

func processOrder(order Order, taxRate float64) float64 {
	total := (calculateItemsTotal(order.Items) - 
		getDiscountAmount(order.DiscountCode)) * (1 + taxRate)

	fmt.Printf("Order %s processed. Total after tax: $%.2f\n", order.ID, total)
	return total
}

func calculateItemsTotal(items []Item) float64 {
	var sum float64
	for _, item := range items {
		sum += item.Price * float64(item.Quantity)
	}
	return sum
}

func getDiscountAmount(discountCode string) float64 {
	if discountCode != "" {
		return 50 // Magic number for discount amount
	}
	return 0
}

func main_order() {
	order := Order{
		ID: "ORD123",
		Items: []Item{
			{Name: "Laptop", Price: 999.99, Quantity: 1},
			{Name: "Mouse", Price: 49.99, Quantity: 2},
			{Name: "Keyboard", Price: 79.99, Quantity: 1},
		},
		DiscountCode: "SUMMER21",
	}
	taxRate := 0.08
	total := processOrder(order, taxRate)
	fmt.Println(total)
}
\ No newline at end of file